crates/contracts/solidity/Makefile

```diff
@@ -1,8 +1,8 @@
 DOCKER := docker
 JQ     := jq
 
-SOLC     := ethereum/solc:0.8.17
-SOLFLAGS := --overwrite --abi --bin --bin-runtime --metadata-hash none --optimize --optimize-runs 1000000
+SOLC     := ethereum/solc:0.8.30
+SOLFLAGS := --overwrite --abi --bin --bin-runtime --metadata-hash none --optimize --optimize-runs 1000000 --evm-version shanghai
```

Comment:
Why Shanghai? Is it to support more chains?

Reply:
There is an existing forked e2e test that runs on a block that only has `shanghai` features. We could theoretically update it but I didn't want to run the risk of sinking a lot of time into that.